sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/SimpleReactiveCosmosRepository.java

```diff
@@ -47,6 +47,7 @@ public SimpleReactiveCosmosRepository(CosmosEntityInformation<T, K> metadata,
 
         CosmosContainerProperties currentProperties = getContainerProperties();
         if (currentProperties != null
+            && entityInformation.isIndexingPolicySpecified()
```

Comment:
Do we also have to check for empty indexing policy?

Reply:
We are not checking for empty indexing policy for now, but future plan is to add an override flag (annotation or to application.properties) - which will ask user to force override indexing policy if they want. If they don't want to override it, then indexing policy will not be updated even if it is used and not same as what they have on their account. 

Regarding IntegrationTest, it is difficult to add one, since for CI, containers are created through spring, and to create a container with indexing policy, the only way to do that is to add the annotation, which if added, will update the indexing policy. 

But we can improve integration tests for this scenario later. For now, the tests in the PR should be sufficient.